### ISSUE TITLE
[java] DU false positive in DataflowAnomalyAnalysis

### ISSUE THREAD

**Issue by Vampire** [pmd/pmd#1682]
**Affects PMD Version:** *6.11.0*
**Rule:** [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis)
**Description:**
Not sure whether this is a duplicate of the others that are open, but at least here another use-case that is failing.
It complains about `logMessage`.

**Code Sample demonstrating the issue:**

``` java
public class Test {
public void append(LogEvent event) {
    String logMessage = wrapIn.wrap(new String(getLayout().toByteArray(event), getLayout().getCharset()));
    DiscordManager discordManager = findDiscordManager();
    if (discordManager == null) {
        bufferMessage(logMessage, event.getTimeMillis());
    } else {
        bufferedMessagesByAppenderName.computeIfPresent(getName(), (name, bufferedMessages) -> {
            bufferedMessages.forEach(bufferedMessage -> sendMessage(discordManager, bufferedMessage));
            return null;
        });
        sendMessage(discordManager, logMessage);
    }
}
}
```

Result:

n/a:2:	Found 'UR'-anomaly for variable 'bufferedMessages' (lines '2'-'9').
n/a:3:	Found 'DU'-anomaly for variable 'logMessage' (lines '3'-'14').

**Running PMD through:** *Gradle*

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.

See also #2647.
